app/models/route.rb

```diff
@@ -3,4 +3,17 @@ class Route < ApplicationRecord
   has_many :bus_stops_routes
   has_many :bus_stops, through: :bus_stops_routes
   default_scope { order :number }
+
+  def stops_by_sequence(route_id, direction)
+    BusStop
+      .joins(:bus_stops_routes)
+      .where(bus_stops_routes: { route_id: route_id, direction: direction })
+      .order('sequence')
+  end
+
+  def next_stop_in_sequence(stop, route_id, direction)
+    stops = stops_by_sequence(route_id, direction)
+    index = stops.find_index(stop) + 1
```

Comment:
I'd suggest to rename this variable. Yes, it's clear that it is an `index`. But why does that matter? Perhaps it could be named `index_of_next_stop`? Or something along those lines.